include/bitcoin/blockchain/interface/block_chain.hpp

```diff
@@ -382,6 +382,17 @@ class BCB_API block_chain
     void remove_mined_txs_from_chosen_list(block_const_ptr blk) override;
 #endif // WITH_MINING
 
+#ifdef WITH_KEOKEN    
+    virtual void fetch_keoken_history(const short_hash& address_hash, size_t limit,
+        size_t from_height, keoken_history_fetch_handler handler) const override;
+
+    virtual void fetch_block_keoken(const hash_digest& hash, bool witness,
+        block_keoken_fetch_handler handler) const override;
+
+    virtual void get_to_keo_transaction(libbitcoin::hash_digest hash,
```

Comment:
This name is confusing; wouldn't "extract_keo_transaction" or "convert_to_keo_transaction" be better?